src/python/pants/backend/shell/lint/shellcheck/subsystem.py

```diff
@@ -20,18 +20,18 @@ class Shellcheck(TemplatedExternalTool):
 
     default_version = "v0.8.0"
     default_known_versions = [
-        "v0.8.0|macos_arm64 |e065d4afb2620cc8c1d420a9b3e6243c84ff1a693c1ff0e38f279c8f31e86634|4049756",
-        "v0.8.0|macos_x86_64|e065d4afb2620cc8c1d420a9b3e6243c84ff1a693c1ff0e38f279c8f31e86634|4049756",
-        "v0.8.0|linux_arm64 |9f47bbff5624babfa712eb9d64ece14c6c46327122d0c54983f627ae3a30a4ac|2996468",
-        "v0.8.0|linux_x86_64|ab6ee1b178f014d1b86d1e24da20d1139656c8b0ed34d2867fbb834dad02bf0a|1403852",
+        "v0.8.0|macos_arm64 |36dffd536b801c8bab2e9fa468163037e0c7f7e0a05298e5ad6299b4dde67e31|14525367",
+        "v0.8.0|macos_x86_64|4e93a76ee116b2f08c88e25011830280ad0d61615d8346364a4ea564b29be3f0|6310442",
+        "v0.8.0|linux_arm64 |8f4810485425636eadce2ec23441fd29d5b1b58d239ffda0a5faf8dd499026f5|4884430",
+        "v0.8.0|linux_x86_64|01d181787ffe63ebb0a2293f63bdc8455c5c30d3a6636320664bfa278424638f|2082242",
     ]
```

Comment:
The shas and sizes have changed. The structure of the packaged dir is different. It is now a tar.gz that contains only the `shellcheck` binary within.

Reply:
For the platforms that are shared, is the shellcheck binary the same? Or does its hash change too?

Reply:
I just now checked, and it turns out that the sha of the binary contained inside the archive _is_ exactly the same for the platforms that are shared. The difference in sha and size of the archive must be because of the difference in compression and/or directory substructure.

slightly formatted results, showing the sha | bytecount of the archive and of the extracted executable:
```
## macos_arm64
https://github.com/vscode-shellcheck/shellcheck-binaries/releases/download/v0.8.0/shellcheck-v0.8.0.darwin.aarch64.tar.gz
36dffd536b801c8bab2e9fa468163037e0c7f7e0a05298e5ad6299b4dde67e31 | 14525367
shellcheck
02dd463e665d0c20a243c600f0d93dd3fd90d20e02355b06660cc18c9da884f5 | 74608983

## macos_x86_64
https://github.com/koalaman/shellcheck/releases/download/v0.8.0/shellcheck-v0.8.0.darwin.x86_64.tar.xz
e065d4afb2620cc8c1d420a9b3e6243c84ff1a693c1ff0e38f279c8f31e86634 | 4049756
shellcheck-v0.8.0/shellcheck
5e0dbf0166a30194f48ec74e453eefaa0807f8150b728281b70a9b91ec2b38da | 32603580

https://github.com/vscode-shellcheck/shellcheck-binaries/releases/download/v0.8.0/shellcheck-v0.8.0.darwin.x86_64.tar.gz
4e93a76ee116b2f08c88e25011830280ad0d61615d8346364a4ea564b29be3f0 | 6310442
shellcheck
5e0dbf0166a30194f48ec74e453eefaa0807f8150b728281b70a9b91ec2b38da | 32603580

## linux_arm64
https://github.com/koalaman/shellcheck/releases/download/v0.8.0/shellcheck-v0.8.0.linux.aarch64.tar.xz
9f47bbff5624babfa712eb9d64ece14c6c46327122d0c54983f627ae3a30a4ac | 2996468
shellcheck-v0.8.0/shellcheck
f0dd2c031db171e6e95fd7b53c6ff32ff96c94e1fb40c5f28a23b50f72f57c89 | 20628560

https://github.com/vscode-shellcheck/shellcheck-binaries/releases/download/v0.8.0/shellcheck-v0.8.0.linux.aarch64.tar.gz
8f4810485425636eadce2ec23441fd29d5b1b58d239ffda0a5faf8dd499026f5 | 4884430
shellcheck
f0dd2c031db171e6e95fd7b53c6ff32ff96c94e1fb40c5f28a23b50f72f57c89 | 20628560

## linux_x86_64
https://github.com/koalaman/shellcheck/releases/download/v0.8.0/shellcheck-v0.8.0.linux.x86_64.tar.xz
ab6ee1b178f014d1b86d1e24da20d1139656c8b0ed34d2867fbb834dad02bf0a | 1403852
shellcheck-v0.8.0/shellcheck
f4bce23c11c3919c1b20bcb0f206f6b44c44e26f2bc95f8aa708716095fa0651 | 6753016

https://github.com/vscode-shellcheck/shellcheck-binaries/releases/download/v0.8.0/shellcheck-v0.8.0.linux.x86_64.tar.gz
01d181787ffe63ebb0a2293f63bdc8455c5c30d3a6636320664bfa278424638f | 2082242
shellcheck
f4bce23c11c3919c1b20bcb0f206f6b44c44e26f2bc95f8aa708716095fa0651 | 6753016
```


Script used to produce the data for the above:
```sh
#!/bin/bash

set -e

OUTPUT_BASE_DIR="gsecrets"

VSCODE_BASE_DIR="${OUTPUT_BASE_DIR}/shellcheck-tests/vscode"
KOALAMAN_BASE_DIR="${OUTPUT_BASE_DIR}/shellcheck-tests/koalaman"

mkdir -p "${VSCODE_BASE_DIR}"
mkdir -p "${KOALAMAN_BASE_DIR}"

download_shellcheck_vscode() {
  version=$1
  platform=$2
  filename="shellcheck-${version}.${platform}.tar.gz"
  URL="https://github.com/vscode-shellcheck/shellcheck-binaries/releases/download/${version}/${filename}"
  echo "$URL"
  curl -s -L -o "${VSCODE_BASE_DIR}/${filename}" "${URL}"
  shasum -a 256 "${VSCODE_BASE_DIR}/${filename}"
  wc -c "${VSCODE_BASE_DIR}/${filename}"

  mkdir -p "${VSCODE_BASE_DIR}/${version}-${platform}"
  tar -xzf "${VSCODE_BASE_DIR}/${filename}" -C "${VSCODE_BASE_DIR}/${version}-${platform}" "shellcheck"
  shasum -a 256 "${VSCODE_BASE_DIR}/${version}-${platform}/shellcheck"
  wc -c "${VSCODE_BASE_DIR}/${version}-${platform}/shellcheck"
}

download_shellcheck_koalaman() {
  version=$1
  platform=$2
  filename="shellcheck-${version}.${platform}.tar.xz"
  URL="https://github.com/koalaman/shellcheck/releases/download/${version}/${filename}"
  echo "$URL"
  curl -s -L -o "${KOALAMAN_BASE_DIR}/${filename}" "${URL}"
  shasum -a 256 "${KOALAMAN_BASE_DIR}/${filename}"
  wc -c "${KOALAMAN_BASE_DIR}/${filename}"

  mkdir -p "${KOALAMAN_BASE_DIR}/${version}-${platform}"
  tar -xzf "${KOALAMAN_BASE_DIR}/${filename}" -C "${KOALAMAN_BASE_DIR}/${version}-${platform}" "shellcheck-${version}/shellcheck"
  shasum -a 256 "${KOALAMAN_BASE_DIR}/${version}-${platform}/shellcheck-${version}/shellcheck"
  wc -c "${KOALAMAN_BASE_DIR}/${version}-${platform}/shellcheck-${version}/shellcheck"
}

echo 'macos_arm64'
download_shellcheck_vscode v0.8.0 darwin.aarch64
echo ''

echo 'macos_x86_64'
download_shellcheck_koalaman v0.8.0 darwin.x86_64
download_shellcheck_vscode v0.8.0 darwin.x86_64

echo 'linux_arm64'
download_shellcheck_koalaman v0.8.0 linux.aarch64
download_shellcheck_vscode v0.8.0 linux.aarch64

echo 'linux_x86_64'
download_shellcheck_koalaman v0.8.0 linux.x86_64
download_shellcheck_vscode v0.8.0 linux.x86_64
```
I also spot-checked the darwin.86 one with plain downloading from the browser and manual running of commands in case I flubbed something up with the script above.

Reply:
Nice! Thanks for verifying it. It's good that they match exactly. 👍